### PR TITLE
Asteroids die and split.

### PR DESCRIPTION
**Pull request by nightblade9**
Some asteroids don't collide properly.

### CODE REVIEW

File: source/PlayState.hx [hunk 1]
```diff
@@ -72,4 +74,41 @@ class PlayState extends HelixState
...
+			var health = Std.int(asteroid.totalHealth / 2);
+			var scale = asteroid.scale.x / 2;
+
+			for (i in 0 ... 2)
+			{
+				// Respawn at half health
+				var newAsteroid = asteroids.recycle(Asteroid);
+				newAsteroid.totalHealth = health;
+				newAsteroid.health = health;
+				asteroids.add(newAsteroid);
```
Comment (NegativeScript) [inline source/PlayState.hx]: Why is `newAsteroid` added to `asteroids`, even though it is acquired from `asteroids.recycle()`?

Comment (nightblade9) [reply source/PlayState.hx]: `recycle` creates a new instance if one isn't available to recycle from the group. Is there a chance we're re-adding the asteroid if it's already in the group? I don't think so (we remove `kill`ed asteroids from the group, right?)

Comment (NegativeScript) [reply source/PlayState.hx]: Thing is, new instances created using `recycle` are automatically added to the group, as discovered upon inspecting the source. The `add` method doesn't do anything if a given object is already in the group, so this is indeed redundant, and harmless

**Comment by NegativeScript** [comment]
Doesn't the `collideResolve` call in the merge conflict belong in the `addAsteroid` method? You know, instead of the placeholder trace message?

**Pull request merged**